File: testers/avro/java-quarkus/generated-and-checked-in/com/example/service/Result.java
package com.example.service;

/** Generic result type - either success value or error */
public sealed interface Result<T, E> permits Result.Ok, Result.Err {
  /** Error result */
  record Err<T, E>(E error) implements Result<T, E> {
    public Err<T, E> withError(E error) {
      return new Err<>(error);
    }
  }

  /** Successful result */
  record Ok<T, E>(T value) implements Result<T, E> {
    public Ok<T, E> withValue(T value) {
      return new Ok<>(value);
    }
  }
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/service/Result.scala
package com.example.service



/** Generic result type - either success value or error */
sealed trait Result[T, E]

object Result {
  /** Error result */
  case class Err[T, E](error: E) extends Result[T, E]

  /** Successful result */
  case class Ok[T, E](value: T) extends Result[T, E]
}